### PR TITLE
update CONTRIBUTING.md to add standard for docstrings

### PR DESCRIPTION
**Pull request by danilito19**
# Summary of Changes

Add documentation for docstrings standards.
(I grepped for """.{70,} to look for 70ish char locations and updated some of them to make them shorter but left some 70-80char lens as they are)

# Related Issues

This completes the first checklist in [this issue](https://github.com/falconry/falcon/issues/1638)

# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [ ] Applied changes to both WSGI and ASGI code paths and interfaces (where applicable).
- [ ] Added **tests** for changed code.
- [ ] Prefixed code comments with GitHub nick and an appropriate prefix.
- [ ] Coding style is consistent with the rest of the framework.
- [ ] Updated **documentation** for changed code.
    - [ ] Added docstrings for any new classes, functions, or modules.
    - [ ] Updated docstrings for any modifications to existing code.
    - [ ] Updated both WSGI and ASGI docs (where applicable).
    - [ ] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [ ] Updated all relevant supporting documentation files under `docs/`.
    - [ ] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [ ] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [ ] Changes (and possible deprecations) have [towncrier](https://towncrier.readthedocs.io/en/actual-freaking-docs/index.html) news fragments under `docs/_newsfragments/`, with the file name format `{issue_number}.{fragment_type}.rst`. (Run `towncrier --draft` to ensure it renders correctly.)

If you have *any* questions to *any* of the points above, just **submit and ask**! This checklist is here to *help* you, not to deter you from contributing!

*PR template inspired by the attrs project.*

### CODE REVIEW

**Comment by danilito19** [comment]
@kgriffs @vytas7 Bringing your attention to this PR. I have a couple of questions.
1. I'm not sure why I can't add reviewers? Is adding reviewers dependent on the pipeline passing? I'm not sure why it didn't pass as my changes are docstrings only.

2. In my humble experience, trying to enforce a new stylistic convention such as this one without an automated test is a recipe for disappointment. As a maintainer myself, I just can't find the mental bandwidth to always remember this. Do we have thoughts on how to automate this check?

**Comment by danilito19** [comment]
> Out of curiosity, is it possible to configure checks to detect line breaks after the opening quotes (as I may happen to accidentally add out of habit)? (IIRC [PEP 257](https://www.python.org/dev/peps/pep-0257/) does not mandate one or another way wrt this aspect.)

@vytas7 I believe this is exactly `D205 | 1 blank line required between summary line and description` which I have added!

For example, currently this fails
```
class InspectVisitor:
    def process(self, instance: _Traversable):
        """Process the instance, by calling the appropriate visit method.
        Uses the `__visit_name__` attribute of the `instance` to obtain the method to use.

```
with `./falcon/inspect.py:545:1: D205 1 blank line required between summary line and description`

File: falcon/vendor/mimeparse/__init__.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 """
-This module wraps code from the MIT-licensed python-mimeparse project. The
-original project source code may be obtained from GitHub by visiting the
+This module wraps code from the MIT-licensed python-mimeparse project.
+
+Theoriginal project source code may be obtained from GitHub by visiting the
```
Comment (vytas7) [inline falcon/vendor/mimeparse/__init__.py:4]: Word rewrapping is a bit off here, _Theoriginal_ (sic) should not be merged.

File: falcon/util/reader.py [hunk 2]
```diff
@@ -370,13 +370,13 @@ def readlines(self, hint=-1):
     # --- implementing IOBase methods, the duck-typing way ---
 
     def readable(self):
-        """Always returns ``True``."""
+        """Return ``True``always."""
```
Comment (vytas7) [inline falcon/util/reader.py:373]: No space after `True`.

File: falcon/request_helpers.py [hunk 3]
```diff
@@ -238,15 +238,15 @@ def _read(self, size, target):
         return target(size)
 
     def readable(self):
-        """Always returns ``True``."""
+        """Return ``True`` always."""
```
Comment (vytas7) [inline falcon/request_helpers.py:241]: Maybe it's just me, but these Return `True`/`False` sound a bit awkward... Could we find an alternative sentence form?

**Review by vytas7 — CHANGES_REQUESTED** [review]
A couple of nitpicks, but this is otherwise getting close to done! :+1:

**Comment by vytas7** [comment]
Something is funky with coverage... I'll try rerunning all CI jobs to check if that makes any difference.

File: falcon/cmd/inspect_app.py [hunk 4]
```diff
@@ -27,7 +27,7 @@
 
 
 def make_parser():
-    """Creates the parsed or the application"""
+    """Create the parsed or the application."""
```
Comment (vytas7) [inline falcon/cmd/inspect_app.py:30]: Create the ~~parser~~ parser or the application.
(But it was this way before, so not required to change in this PR)

File: falcon/util/reader.py [hunk 5]
```diff
@@ -370,13 +370,13 @@ def readlines(self, hint=-1):
...
         return True
 
     def seekable(self):
-        """Always returns ``False``."""
+        """Return ``False``always."""
         return False
 
     def writeable(self):
-        """Always returns ``False``."""
+        """Return ``False``always."""
```
Comment (CaselIT) [inline falcon/util/reader.py:381]: Also missing space here

File: falcon/util/reader.py [hunk 6]
```diff
@@ -370,13 +370,13 @@ def readlines(self, hint=-1):
     # --- implementing IOBase methods, the duck-typing way ---
 
     def readable(self):
-        """Always returns ``True``."""
+        """Return ``True``always."""
         return True
 
     def seekable(self):
-        """Always returns ``False``."""
+        """Return ``False``always."""
```
Comment (CaselIT) [inline falcon/util/reader.py:377]: Also missing space here

**Review by CaselIT — CHANGES_REQUESTED** [review]
just a couple additional changes.

Regarding this check, I think we could have another tox env to run these checks, and anoter entry in the text matrix in the workflow action

File: falcon/request_helpers.py (hunk 3 above)
Comment (danilito19) [reply falcon/request_helpers.py:241]: I'd love that, but I banged my head against a dictionary (a physical one, not a python one) trying to reword this. It needs to be in imperative mood. I could take out the "always"?

**Comment by danilito19** [comment]
@CaselIT silly me forgot to add to actions. it's now there `Run tests / tox -e pep8-docstrings (3.8 on ubuntu-20.04) (pull_request) `

File: falcon/vendor/mimeparse/__init__.py (hunk 1 above)
Comment (danilito19) [reply falcon/vendor/mimeparse/__init__.py]: Unsure if you wanted a specific change for the word wrapping. I did update `Theorigin..` > `The original`

**Comment by danilito19** [comment]
This this project now have a Rerun actions option? as shown [here[(https://stackoverflow.com/questions/56435547/how-do-i-re-run-github-actions)

I'm trying to re-run the failing actions because I"m unsure why they even failed, but can't find the button anywhere.

**Comment by CaselIT** [comment]
> 
> 
> This this project now have a Rerun actions option? as shown [here[(https://stackoverflow.com/questions/56435547/how-do-i-re-run-github-actions)
> 
> I'm trying to re-run the failing actions because I"m unsure why they even failed, but can't find the button anywhere.

It does, but I don't think it's available outside the member of the repository

**Review by CaselIT — APPROVED** [review]
Looks ok.

**Comment by vytas7** [comment]
I'm not sure if it is the latest changes aggravating (or just checking failures more carefully) it, but it seems Daphne is failing even more often than before...

**Comment by CaselIT** [comment]
It seems that the py3.5.2 is broken. It seems to be an outdated setuptools. Maybe updating it makes it work again?

File: falcon/request_helpers.py (hunk 3 above)
Comment (vytas7) [reply falcon/request_helpers.py:241]: Does the imperative need to be first in that `flake8` check?
Maybe just "Always return ``True``" if that is accepted?

Comment (vytas7) [reply falcon/request_helpers.py:241]: Hm, could we possibly add a `noqa: D401` for these? :thinking:

Comment (CaselIT) [reply falcon/request_helpers.py:241]: How about `Return ``True`` in call cases` or `Return ``True`` in every case`?

Comment (vytas7) [reply falcon/request_helpers.py:241]: That could also work.

Personally, I vote for simply adding `# noqa: D401` in cases like this.
The docstring _is_ in imperative mood, just the tool is not advanced enough (Deep Learning + [NLP](https://en.wikipedia.org/wiki/Natural_language_processing) patches to the tool anyone? :smiling_imp:).

Comment (danilito19) [reply falcon/request_helpers.py:241]: Thanks for the suggestions! Please let me know which you prefer as I don't have an opinion on this (it's pretty minor IMHO).

Comment (CaselIT) [reply falcon/request_helpers.py:241]: > (Deep Learning + [NLP](https://en.wikipedia.org/wiki/Natural_language_processing) patches to the tool anyone? 😈).

Just a whitelist in this case is way easier

Comment (vytas7) [reply falcon/request_helpers.py:241]: @kgriffs any preference on attempting to reword at any cost, or simply adding exceptions where the tool is not smart enough?

I'm fine with either.

**Review by vytas7 — APPROVED** [review]
Let's merge this as-is; it is dragging on for no reason. The contentious part needing decision is small, and it's not the end of the world: we can fix it in a follow-up PR.

:tada:

**Pull request merged**